gdal/swig/python/osgeo/utils/pct2rgb.py

```diff
@@ -211,22 +166,19 @@ def main(argv):
             band_lookup = lookup[iBand]
 
             dst_data = Numeric.take(band_lookup, src_data)
-            tif_ds.GetRasterBand(iBand + 1).WriteArray(dst_data, 0, iY)
+            dst_ds.GetRasterBand(iBand + 1).WriteArray(dst_data, 0, iY)
 
         progress((iY + 1.0) / src_ds.RasterYSize)
 
-
-    tif_ds = None
```

Comment:
I think this should be kept. Otherwise the dataset might not be properly finalized before gdal.Open() is run
See
```
>>> from osgeo import gdal
>>> ds = gdal.GetDriverByName('GTiff').Create('/vsimem/foo.tif', 1, 1)
>>> ds = gdal.Open('/vsimem/foo.tif')
ERROR 4: `/vsimem/foo.tif' not recognized as a supported file format.
>>> 
>>> ds = gdal.GetDriverByName('GTiff').Create('/vsimem/foo.tif', 1, 1)
>>> ds = None
>>> ds = gdal.Open('/vsimem/foo.tif')
```

Reply:
Or at least dst_ds = None should be moved before the gdal.Open() call in the if branch

Reply:
I wanted to copy tif_ds to dst_ds without reading it from file, please check if I took the right approach in the next commit. thanks!